File: frontend/src/types/project.ts
export interface Project {
  id: number
  name: string
  description: string
  techStack: string | null
  repoUrl: string | null
  demoUrl: string | null
  published: boolean
  authorId?: number
  coverUrl?: string | null
  createdAt: string
  updatedAt: string
  deletedAt?: string | null
}

  export interface ProjectCreateRequest {
    name: string
    description: string
    techStack?: string
    repoUrl?: string
    demoUrl?: string
    published?: boolean
  }

  export interface ProjectUpdateRequest {
    name: string
    description: string
    techStack?: string
    repoUrl?: string
    demoUrl?: string
    published: boolean
  }

File: backend/src/main/java/com/zzh/personal_hub/project/entity/Project.java
package com.zzh.personal_hub.project.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

import java.time.Instant;

@Getter
@Setter
@Entity
@Table(name = "project")
public class Project {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 200)
    private String name;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String description;

    /** 技术栈，逗号分隔，如 React,Spring Boot,MySQL */
    @Column(length = 500)
    private String techStack;

    @Column(length = 500)
    private String repoUrl;

    @Column(length = 500)
    private String demoUrl;

    @Column(nullable = false)
    private Boolean published = false;

    @Column(nullable = false, updatable = false)
    private Instant createdAt = Instant.now();

    @Column(nullable = false)
    private Instant updatedAt = Instant.now();
}